lib/ProductOpener/APITest.pm

```diff
@@ -179,4 +183,112 @@ sub construct_test_url ($target, $prefix) {
 	return $url;
 }
 
+=head2 execute_api_tests($file, $tests_ref)
+
+Initialize tests and execute them.
+
+=head3 Arguments
+
+=head4 $file test file name
+
+The *.t test files call execute_api_tests() with _FILE_ as the first parameter,
+and the directories for the tests are derived from it.
+
+=head4 $tests_ref reference to list of tests
+
+The tests are in a structure like this:
+
+my $tests_ref = (
+    [
+		{
+			test_case => 'no-body',
+			method => 'POST',		# defaults to GET
+			subdomain => 'world',	# defaults to "world"
+			path => '/api/v3/product/12345678',
+			query_string => '?some_param=some_value&some_other_param=some_other_value'	# optional
+			form => { field_name => field_value, .. },	# optional, will not be sent if there is a body
+			body => '{"some_json_field": "some_value"}',	# optional
+		}
+    ],
+);
+
+=cut
+
+sub execute_api_tests ($file, $tests_ref) {
+
+	my ($test_id, $test_dir, $expected_result_dir, $update_expected_results) = (init_expected_results($file));
+
+	my $ua = LWP::UserAgent->new();
+
+	foreach my $test_ref (@$tests_ref) {
+		my $test_case = $test_ref->{test_case};
+		my $url = construct_test_url($test_ref->{path} . ($test_ref->{query_string} || ''),
+			$test_ref->{subdomain} || 'world');
+
+		my $method = $test_ref->{method} || 'GET';
+
+		my $response;
+
+		# Send the request
+		if ($method eq 'GET') {
+			$response = $ua->get($url);
+		}
+		elsif ($method eq 'POST') {
+			if (defined $test_ref->{body}) {
+				$response = $ua->post(
+					$url,
+					Content => encode_utf8($test_ref->{body}),
+					"Content-Type" => "application/json; charset=utf-8"
+				);
+			}
+			elsif (defined $test_ref->{form}) {
+				$response = $ua->post($url, Content => $test_ref->{form});
+			}
+			else {
+				$response = $ua->post($url);
+			}
+		}
+
+		# Check if the request was successful
+		if (not $response->is_success) {
```

Comment:
We could also test a  $response has a different return value, so I would instead handle a $test_ref->code parameter (and default to is_success)

Reply:
Good idea, will do that.